src/data_access/FileUserDataAccessObject.java

```diff
@@ -0,0 +1,92 @@
+package data_access;
+
+import entity.*;
+
+import java.io.*;
+import java.util.*;
+
+public class FileUserDataAccessObject {
+    private final File usersFile;
+
+    // contains the content in each column
+    private final Map<String, Integer> headers = new LinkedHashMap<>();
+
+    // mapping username to User
+    private final Map<String, User> accounts = new HashMap<>();
+
+    private UserFactory userFactory;
+
+    public FileUserDataAccessObject(String csvPath, UserFactory userFactory) throws IOException {
+        this.userFactory = userFactory;
+        this.usersFile = new File(csvPath);
+        headers.put("username", 0);
```

Comment:
I think we should use a different delimiter instead of commas for the csv because one column can contain several commas in itself. For example, a user's bio may have multiple commas.